### PR TITLE
Feature/gui parameters interface & constant_oil_sources

### PR DESCRIPTION
**Pull request by Herdran**
Implementation of gui interface for simulation parameters setting, currently lacking functionality of adding constant oil spill as well as manualy setting coordinates using map view
image

### CODE REVIEW

**Comment by Herdran** [comment]
Image preview is not readable due to coordinates values being very close to each other which means that with the image currently used it would't even show a square of 1px/1px, which is why it has its borders artificially expanded to at least show anything for presentation purposes. It might be worth considering using higher quality image, like the original blue marble one that has a resolution of 43200 x 21600 px, but it also weights around 0.5GB which is why I didn't decide to do it as of right now

File: src/constatnts.py [hunk 1]
```diff
@@ -3,57 +3,70 @@
...
 
-SIMULATION_TIME = None
+    SIMULATION_INITIAL_PARAMETERS = None
+    SIMULATION_TIME = None
 
-def set_simulation_coordinates_parameters(TOP_COORD, DOWN_COORD, LEFT_COORD, RIGHT_COORD):
-    global POINT_LAT_SIZE, POINT_LON_SIZE, POINT_LAT_CENTERS, POINT_LON_CENTERS, SIMULATION_INITIAL_PARAMETERS, SIMULATION_TIME
 
-    SIMULATION_INITIAL_PARAMETERS = SimulationRunParameters(
+def set_simulation_coordinates_parameters(top_coord,
```
Comment (bblachut) [inline src/constatnts.py]: type hints

**Review by bblachut — COMMENTED** [review]
Ok, but type hints are missing. Mainly in the gui.py file, outside I left a comment

File: src/constatnts.py (hunk 1 above)
Comment (Mapet13) [reply src/constatnts.py]: also names of arguments should not be in UPPER case - in python that suggest the constatnts

File: src/gui.py [hunk 2]
```diff
@@ -1,17 +1,22 @@
 import logging
+import os.path
 import tkinter as tk
-from tkinter import DISABLED, NORMAL
+from tkinter import DISABLED, NORMAL, filedialog
+import re
 
 import numpy as np
 import threading
+
```
Comment (Mapet13) [inline src/gui.py]: imports could be sorted

File: src/gui.py [hunk 3]
```diff
@@ -622,21 +576,587 @@ def get_data_processor() -> DataProcessor:
...
+                self.map_view_frame = tk.Frame(inputs_frame)
+                self.map_view_frame.grid(row=0, column=0, rowspan=4, padx=3, pady=3, sticky=tk.N + tk.S)
+
+                self.map_view = tk.Canvas(self.map_view_frame)
+                self.load_and_crop_image()
+
+                self.map_view.grid(row=0, column=0, rowspan=3, padx=3, pady=3, sticky=tk.N + tk.S)
+
+                # self.manual_map_coords_selection = tk.Button(self.map_view_frame, text='manual_coords_selection TODO',
+                #                                              command=self.manual_map_coords_selection)
```
Comment (Mapet13) [inline src/gui.py]: dont leave comments

File: src/gui.py [hunk 4]
```diff
@@ -622,21 +576,587 @@ def get_data_processor() -> DataProcessor:
...
+                # self.manual_map_coords_selection.grid(row=4, column=0, padx=3, pady=3, sticky=tk.N + tk.S)
+
+                neighborhood_label = tk.Label(neighborhood_type_frame,
+                                              text="Neighborhood type:",
+                                              font=("Arial", 14, "bold"), padx=3, pady=3)
+                neighborhood_label.grid(row=0, column=0, rowspan=1, padx=3, pady=3, sticky=tk.N + tk.S)
+
+                self.neighborhood_var = tk.IntVar()
+                NM = tk.Radiobutton(neighborhood_type_frame, text="Moore", variable=self.neighborhood_var, value=0)
```
Comment (Mapet13) [inline src/gui.py]: upper case suggest constatnts

File: src/gui.py [hunk 5]
```diff
@@ -622,21 +576,587 @@ def get_data_processor() -> DataProcessor:
...
+                        self.down_coord_validation_label.config(text="Valid value")
+                        self.correctly_set_parameters[1] = 1
+                        self.check_all_parameters_validity_and_refresh_image()
+                        return True
+                    self.down_coord_validation_label.config(text="Invalid value")
+                    self.confirm_and_continue.config(state=DISABLED)
+                    self.correctly_set_parameters[1] = 0
+                return False
+
+            def validate_coordinates_left(self):
```
Comment (Mapet13) [inline src/gui.py]: DRY!!!!!!!!!!!! - there is a lot of copy-paste code
maybe not in that PR, but we should think about cleaning that

**Pull request merged**